### PR TITLE
Update Sample App Edge View

### PR DESCRIPTION
**Pull request by cacheung**
Update Sample App Edge View with the latest sample app format.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: example/ios/Podfile.lock [hunk 1]
```diff
@@ -2,22 +2,22 @@ PODS:
   - AEPAssurance (3.0.1):
     - AEPCore (>= 3.1.0)
     - AEPServices (>= 3.1.0)
-  - AEPCore (3.7.0):
+  - AEPCore (3.4.2):
```
Comment (emdobrin) [inline example/ios/Podfile.lock]: this got reverted, can you run pod update again?

File: example/lib/assurance.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_aepassurance/flutter_aepassurance.dart';
+
+class AssurancePage extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<AssurancePage> {
```
Comment (emdobrin) [inline example/lib/assurance.dart:20]: is this a merge from the consent branch and if so can you indicate what files are new with this PR?

File: example/lib/edge.dart [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import 'dart:developer';
```
Comment (cacheung) [inline example/lib/edge.dart]: This edge.dart file is new for sample app, other files are reviewed in the consent sample app PR.

File: example/lib/assurance.dart (hunk 2 above)
Comment (cacheung) [reply example/lib/assurance.dart:20]: edge.dart is new.

File: example/lib/main.dart [hunk 4]
```diff
@@ -1,434 +1,67 @@
...
-    ));
+            ElevatedButton(
+              child: const Text('ASSURANCE'),
+              onPressed: () {
+                Navigator.push(context,
+                    MaterialPageRoute(builder: (context) => AssurancePage()));
+              },
+            ),
+            ElevatedButton(
+              child: const Text('EDGE'),
```
Comment (cacheung) [inline example/lib/main.dart:58]: New: Add EdgePage button from the main.dart

File: example/android/app/build.gradle [hunk 5]
```diff
@@ -25,7 +25,7 @@ apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 32
+    compileSdkVersion 31
```
Comment (addb) [inline example/android/app/build.gradle]: Why did we make this change?

File: example/lib/edge.dart [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    Map<dynamic, dynamic> xdmData = {"eventType": "SampleEventType"};
+    Map<String, dynamic> data = {"free": "form", "data": "example"};
+
+    final ExperienceEvent experienceevent = ExperienceEvent({
+      "xdmData": xdmData,
+      "data": data,
+      "datasetIdentifier": datasetId,
+    });
+    try {
+      result = await Edge.sendEvent(experienceevent);
```
Comment (addb) [inline example/lib/edge.dart]: Can you paste the event sent by this API call, so we can verify the implementationDetails like version attached in the request to konductor is correct.

File: example/pubspec.lock [hunk 7]
```diff
@@ -82,7 +82,7 @@ packages:
       path: "../plugins/flutter_aepedge"
       relative: true
     source: path
-    version: "1.0.0"
+    version: "0.0.0"
```
Comment (addb) [inline example/pubspec.lock]: version 0.0.0 ?

File: example/android/app/build.gradle (hunk 5 above)
Comment (cacheung) [reply example/android/app/build.gradle:28]: We set to 31 for other plugins as well. It was mistyped for 32 earlier so change it back to 31 for consistency.

File: example/lib/edge.dart (hunk 6 above)
Comment (cacheung) [reply example/lib/edge.dart]: {
	"meta": {
		"konductorConfig": {
			"streaming": {
				"enabled": true,
				"recordSeparator": "\u0000",
				"lineFeed": "\n"
			}
		},
		"state": {
			"entries": [{
				"key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_cluster",
				"value": "or2",
				"maxAge": 1800
			}, {
				"key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_identity",
				"value": "CiYzMTYyOTc0MzgzNTEwMTQzNTI3NjUzNzQwNDc1ODAyNjkzOTE3MlIOCJn9lo-lMBgBKgNPUjLwAZn9lo-lMA==",
				"maxAge": 34128000
			}]
		}
	},
	"xdm": {
		"identityMap": {
			"ECID": [{
				"primary": false,
				"id": "31629743835101435276537404758026939172",
				"authenticatedState": "ambiguous"
			}]
		},
		"implementationDetails": {
			"version": "3.4.2+1.3.0",
			"name": "https:\/\/ns.adobe.com\/experience\/mobilesdk\/ios\/flutter",
			"environment": "app"
		}
	},
	"events": [{
		"xdm": {
			"_id": "92464BE0-1DC6-4DE5-913D-02B9814B700A",
			"eventType": "SampleEventType",
			"timestamp": "2022-07-31T01:48:02Z"
		},
		"data": {
			"free": "form",
			"data": "example"
		}
	}]
}

File: example/android/app/build.gradle (hunk 5 above)
Comment (cacheung) [reply example/android/app/build.gradle]: It was always set to 31, there was a typo and changed to 32.  Now changed it back.

File: example/lib/edge.dart (hunk 6 above)
Comment (cacheung) [reply example/lib/edge.dart]: Android:


 {
      "xdm": {
        "implementationDetails": {
          "environment": "app",
          "name": "https:\/\/ns.adobe.com\/experience\/mobilesdk\/android\/flutter",
          "version": "1.11.1+1.3.2"
        },
        "identityMap": {
          "ECID": [
            {
              "id": "12749448659673225666642179032797304370",
              "authenticatedState": "ambiguous",
              "primary": false
            }
          ]
        }
      },
      "meta": {
        "state": {
          "entries": [
            {
              "maxAge": 1800,
              "value": "va6",
              "key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_cluster"
            },
            {
              "maxAge": 34128000,
              "value": "CiYxMjc0OTQ0ODY1OTY3MzIyNTY2NjY0MjE3OTAzMjc5NzMwNDM3MFIOCK3jnqCgMBgBKgNPUjLwAdbngtulMA==",
              "key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_identity"
            }
          ]
        },
        "konductorConfig": {
          "streaming": {
            "lineFeed": "\n",
            "enabled": true,
            "recordSeparator": "\u0000"
          }
        }
      },
      "events": [
        {
          "xdm": {
            "eventType": "SampleEventType",
            "_id": "676741e3-873d-46d3-9fa2-38804f621ee7",
            "timestamp": "2022-08-01T21:48:01.745Z"
          },
          "data": {
            "data": "example",
            "free": "form"
          }
        }
      ]
    }

File: example/lib/edge.dart [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    if (!mounted) return;
+
+    setState(() {
+      _edgeVersion = edgeVersion;
+    });
+  }
+
+  Future<void> sendEvent([datasetId]) async {
+    late List<EventHandle> result;
+    Map<dynamic, dynamic> xdmData = {"eventType": "SampleEventType"};
```
Comment (emdobrin) [inline example/lib/edge.dart]: this should be string, dynamic too

File: example/lib/edge.dart [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    final ExperienceEvent experienceevent = ExperienceEvent({
+      "xdmData": xdmData,
+      "data": data,
+      "datasetIdentifier": datasetId,
+    });
+    try {
+      result = await Edge.sendEvent(experienceevent);
+    } on PlatformException {
+      log("Failed to send experience event");
+    }
```
Comment (emdobrin) [inline example/lib/edge.dart]: I think we can get this PlatformException removed with the changes in https://github.com/adobe/aepsdk_flutter/pull/3#discussion_r932959640

File: example/lib/edge.dart [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+  // Platform messages are asynchronous, so we initialize in an async method.
+  Future<void> initPlatformState() async {
+    late String edgeVersion;
+    // Platform messages may fail, so we use a try/catch PlatformException.
+    try {
+      edgeVersion = await Edge.extensionVersion;
+    } on PlatformException {
+      log("Failed to get extension versions");
+    }
```
Comment (emdobrin) [inline example/lib/edge.dart]: I don't see an error being thrown for the extensionVersion API, can we remove the try/catch here too?

Comment (cacheung) [reply example/lib/edge.dart]: removed

**Review by emdobrin — APPROVED** [review]

**Pull request merged**